packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/model/configurations/ModelConfiguration.java

```diff
@@ -0,0 +1,49 @@
+package software.uncharted.terarium.hmiserver.models.dataservice.model.configurations;
+
+import com.fasterxml.jackson.annotation.JsonManagedReference;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Transient;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.experimental.Accessors;
+import software.uncharted.terarium.hmiserver.annotations.TSIgnore;
+import software.uncharted.terarium.hmiserver.annotations.TSModel;
+import software.uncharted.terarium.hmiserver.models.TerariumAsset;
+
+@EqualsAndHashCode(callSuper = true)
+@Data
+@TSModel
+@Accessors
+@Entity
+public class ModelConfiguration extends TerariumAsset {
+	private UUID calibrationRunId;
+	private UUID modelId;
+
+	@Transient
+	private Map<String, Semantic> values;
+
+	@TSIgnore
+	@OneToMany(mappedBy = "modelConfiguration", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
+	@Schema(accessMode = Schema.AccessMode.READ_ONLY)
+	@JsonManagedReference
+	private List<ObservableSemantic> observableSemanticList;
+
+	@TSIgnore
+	@OneToMany(mappedBy = "modelConfiguration", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
+	@Schema(accessMode = Schema.AccessMode.READ_ONLY)
+	@JsonManagedReference
+	private List<ParameterSemantic> parameterSemanticList;
+
+	@TSIgnore
+	@OneToMany(mappedBy = "modelConfiguration", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
+	@Schema(accessMode = Schema.AccessMode.READ_ONLY)
+	@JsonManagedReference
+	private List<InitialSemantic> initialSemanticList;
```

Comment:
I may have missed a chat here. So we have these three `List<blah>` members that seem to be used as proxies to populate "values".

- Why are we doing this?

So then the UI is working with `ModelConfiguration.values` which is a map, but the API, according to swagger, is expecting the ModelConfiguration to be specified in terms of these three xyzSemanticList.

Can someone maybe shed some light here?

Reply:
So the plan for the class is to have the `values` map. However this won't serialize correctly unless its all just one giant JSON blob, so the individual lists of types are needed to maintain the foreign-key relationship between the model config and the various Semantic types.  The lists were @TSIgnored to keep them hidden from the front end and instead use the map
If we want to use the lists throughout though that would make things easier!